### ISSUE TITLE
retain face id

### ISSUE THREAD

**Issue by bryanhpchiang** [g-luo/dual_process#1]
super cool work

i looked at the demo_optimize nb and it seems like the for the smiling example the face changes / loses consistency? ie its not the same person anymore

**Comment by g-luo** [comment]
Thanks for checking out our repo @bryanhpchiang! 

The face changes because we don't have any losses to ensure consistency to the original seed. Editing / consistency isn't the goal of our work; rather, our goal is to satisfy the control question purely using gradients from the VLM. If you care about consistency, you might want to look into optimizing with a lower learning rate, adding an additional loss term for this, or turning on the LoRA only in later denoising steps ([a trick used by Concept Sliders](https://github.com/rohitgandikota/sliders)).

**Comment by g-luo** [comment]
Yes! You can use a cosine distance between face embeddings, like from [FaceNet](https://github.com/timesler/facenet-pytorch). [Universal Guidance](https://github.com/arpitbansal297/Universal-Guided-Diffusion) explores this kind of loss on Stable Diffusion.

I also added some more documentation to demo_optimize.ipynb; a simple way to get more visual consistency with our method is by training on a single seed instead of multiple (by setting `config["opt_kwargs"]["train_n"]=1` instead of the default value). The demo results we show that look more visually consistent are trained on a single seed.
